### PR TITLE
reworking networking and partition of nodes.  added context manager.

### PR DESCRIPTION
**Pull request by kensipe**
I love that it allows me to do this:

```
        with iptable_rules(host):
            block_port(host, port)
            time.sleep(7)
```

also possible is:

```
           with disconnected_master():
            # blah
```

### CODE REVIEW

**Comment by sschneid** [comment]
I'm 👍  -- @mohitsoni do you want to review this since it affects `dcos/master.py`?

**Comment by kensipe** [comment]
@nickbp may want to as well... /cc: @mohitsoni There were a number of approaches for partitioning with differences between master and agents.  This consolidates that plus it provides context manager options.   I believe I left the master API intact and with similar behavior.  The strong difference is that there was 1 ssh invoked command for a number of activities for the master  in the past which are broken to be discreet.    (save iptables, flush, open it wide).    There were odd difference in master vs agents on iptable saved file that didn't seem necessary or useful.

I have found that I have desires to save iptables and make a small change vs a major change.  This change makes it nice to easily allow that.

**Comment by kensipe** [comment]
I rebased

File: shakedown/dcos/network.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
+"""Utilities for working with nodes
+    There are a number of common utilies for working with agents and master nodes
+    which are provided here.
+"""
```
Comment (sschneid) [inline shakedown/dcos/network.py]: These should all be documented in [API.md](https://github.com/dcos/shakedown/blob/master/API.md).

**Pull request merged**